pkger/service.go

```diff
@@ -1553,40 +1541,53 @@ func (s *Service) applyLabels(labels []*label) applier {
 		},
 		rollbacker: rollbacker{
 			resource: resource,
-			fn:       func(_ influxdb.ID) error { return s.rollbackLabels(rollBackLabels) },
+			fn:       func(_ influxdb.ID) error { return s.rollbackLabels(ctx, rollBackLabels) },
 		},
 	}
 }
 
-func (s *Service) rollbackLabels(labels []*label) error {
-	var errs []string
-	for _, l := range labels {
-		if l.existing == nil {
-			err := s.labelSVC.DeleteLabel(context.Background(), l.ID())
-			if err != nil {
-				errs = append(errs, l.ID().String())
-			}
-			continue
+func (s *Service) rollbackLabels(ctx context.Context, labels []*label) error {
+	rollbackFn := func(l *label) error {
+		var err error
+		switch {
+		case l.shouldRemove:
```

Comment:
Took me a sec to realize `shouldRemove` causes a Create and `existing == nil` causes a delete because this is a rollback operation :)